src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -713,15 +713,15 @@ public void Insert(IEnumerable<Block> blocks)
             }
         }
 
-        private AddBlockResult Suggest(Block? block, BlockHeader header,
-            BlockTreeSuggestOptions options = BlockTreeSuggestOptions.ShouldProcess,
-            bool? setAsMain = null)
+        private AddBlockResult Suggest(Block? block, BlockHeader header, BlockTreeSuggestOptions options = BlockTreeSuggestOptions.ShouldProcess)
         {
             bool shouldProcess = (options & BlockTreeSuggestOptions.ShouldProcess) != 0;
             bool fillBeaconBlock = (options & BlockTreeSuggestOptions.FillBeaconBlock) != 0;
-            if (_logger.IsTrace)
-                _logger.Trace(
-                    $"Suggesting a new block. BestSuggestedBlock {BestSuggestedBody}, BestSuggestedBlock TD {BestSuggestedBody?.TotalDifficulty}, Block TD {block?.TotalDifficulty}, Head: {Head}, Head TD: {Head?.TotalDifficulty}, Block {block?.ToString(Block.Format.FullHashAndNumber)}. ShouldProcess: {shouldProcess}, TryProcessKnownBlock: {fillBeaconBlock}");
+            bool setAsMainSet = (options & BlockTreeSuggestOptions.SetAsMain) != 0;
```

Comment:
it is weird that we have SetAsMain and DontSetAsMain option

Reply:
yeah, but before we had bool? setAsMain = null. We could force to set or not set, or derive that from!ShouldProcess flag.  Left it the same here. Otherwise, we can change it for ShouldProcess contain flag NotSetAsMain and SetAsMain being the default? It's riskier in my opinion.

Reply:
Shoudve kept as bool or make the option a proper struct imo.

Reply:
why?

Reply:
Bit manipulation is overly complicated for something that could just use a standard struct, hard to read. There is no noticable performance nor size advantage for this use case.